File: java/lab4/app/src/main/java/com/practice/ordersys/MainActivity.java
package com.practice.ordersys;

import androidx.activity.result.ActivityResult;
import androidx.activity.result.ActivityResultCallback;
import androidx.activity.result.ActivityResultLauncher;
import androidx.activity.result.contract.ActivityResultContracts;
import androidx.appcompat.app.AppCompatActivity;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;

import com.practice.ordersys.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {
    private ActivityMainBinding binding;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        ActivityResultLauncher<Intent> resultLauncher = registerForActivityResult(
                new ActivityResultContracts.StartActivityForResult(),
                (ActivityResultCallback<ActivityResult>) result -> {
                    if (result.getResultCode() == Activity.RESULT_OK) {
                        Bundle b = result.getData().getExtras();
                        binding.tvMeal.setText(String.format(
                            "飲料： %s\n\n" + "甜度： %s\n\n" + "冰塊： %s",
                                b.getString("drink"),
                                b.getString("sugar"),
                                b.getString("ice"))
                        );
                    }
                });

        binding.btnChoice.setOnClickListener(view -> {
            Intent intent = new Intent(this, SecActivity.class);
            resultLauncher.launch(intent);
        });
        }
}

File: kotlin/lab4/app/src/main/java/com/practice/ordersys/MainActivity.kt
package com.practice.ordersys

import android.app.Activity
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import androidx.activity.result.ActivityResult
import androidx.activity.result.contract.ActivityResultContracts
import com.practice.ordersys.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        val resultLauncher =
            registerForActivityResult(
                ActivityResultContracts.StartActivityForResult()){ result ->
                    if(result.resultCode == Activity.RESULT_OK){
                        val myData: Intent? = result.data
                }
            }
        binding.btnChoice.setOnClickListener{
            val intent = Intent(this, SecActivity::class.java)
            resultLauncher.launch(intent)
        }
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        data?.extras?.let {
            if (resultCode == Activity.RESULT_OK){
                binding.tvMeal.text =
                    "飲料： ${it.getString("drink")}\n\n" +
                    "甜度： ${it.getString("sugar")}\n\n" +
                    "冰塊： ${it.getString("ice")}"
            }
        }
    }
}